code/game/objects/items/wayfinding.dm

```diff
@@ -1,32 +1,138 @@
 /obj/machinery/pinpointer_dispenser
-	name = "wayfinding pinpointer dispenser"
-	icon = 'icons/obj/bureaucracy.dmi'
-	icon_state = "ticketmachine"
-	desc = "Having trouble finding your way? This machine dispenses pinpointers that point to common locations."
+	name = "wayfinding pinpointer synthesizer"
+	icon = 'icons/obj/machines/wayfinding.dmi'
+	icon_state = "pinpointersynth"
+	desc = "Having trouble finding your way? This machine synthesizes pinpointers that point to common locations."
 	density = FALSE
 	layer = HIGH_OBJ_LAYER
-	var/list/obj/item/pinpointer/wayfinding/pinpointers = list()
-	var/spawn_cooldown = 1200 //deciseconds per person to spawn another pinpointer
+	var/list/user_spawn_cooldowns = list()
+	var/list/user_interact_cooldowns = list()
+	var/spawn_cooldown = 5 MINUTES //time per person to spawn another pinpointer
+	var/interact_cooldown = 20 SECONDS //time per person for subsequent interactions
+	var/start_bal = 200 //how much money it starts with to cover wayfinder refunds
+	var/refund_amt = 40 //how much money recycling a pinpointer rewards you
+	var/datum/bank_account/synth_acc = new /datum/bank_account/remote
+	var/ppt_cost = 65 //Jan 6 '20: Assistant can buy one roundstart (125 cr starting)
+	var/expression_timer
+
+/obj/machinery/pinpointer_dispenser/Initialize(mapload)
+	..()
+	var/datum/bank_account/civ_acc = SSeconomy.get_dep_account(ACCOUNT_CIV)
+	if(civ_acc)
+		synth_acc.transfer_money(civ_acc, start_bal) //float has to come from somewhere, right?
+
+	synth_acc.account_holder = name
+
+	desc += " Only [ppt_cost] credits! It also likes making costumes."
+
+	set_expression("neutral")
 
 /obj/machinery/pinpointer_dispenser/attack_hand(mob/living/carbon/user)
-	if(world.time < pinpointers[user.real_name])
-		var/secsleft = (pinpointers[user.real_name] - world.time) / 10
-		to_chat(user, "<span class='warning'>You need to wait [secsleft/60 > 1 ? "[round(secsleft/60)] minute\s" : "[round(secsleft)] second\s"].</span>")
+	if(world.time < user_interact_cooldowns[user.real_name])
+		to_chat(user, "<span class='warning'>It doesn't respond.</span>")
+		return
+
+	user_interact_cooldowns[user.real_name] = world.time + interact_cooldown
+
+	for(var/obj/item/pinpointer/wayfinding/WP in user.GetAllContents())
```

Comment:
if(var/obj/item/pinpointer/wayfinding/WP in user.GetAllContents())
You wont be running this more than once

Reply:
if(var/obj/item/pinpointer/wayfinding in user.GetAllContents())

Reply:
Doesn't work, unfortunately. Neither does plain /obj/item/pinpointer/wayfinding.  But as you say, it'll only run until it finds a pinpointer. The for loop will terminate on the return, won't it?